lcls_tools/common/devices/yaml/metadata.py

```diff
@@ -1,18 +1,29 @@
 from typing import List
 
 
-def get_magnet_metadata(magnet_names: List[str] = []):
+def get_magnet_metadata(magnet_names: List[str] = [], method: callable = None, **kwargs):
     # return a data structure of the form:
     # {
     #  mag-name-1 : {metadata-field-1 : value-1, metadata-field-2 : value-2},
     #  mag-name-2 : {metadata-field-1 : value-1, metadata-field-2 : value-2},
     #  ...
     # }
-    if magnet_names:
-        raise NotImplementedError(
-            "No method of getting additional metadata for magnets."
-        )
-    return {}
+    if magnet_names and method:
+        # Add any additional metadata fields here
```

Comment:
What would method be here?

Reply:
Method is the method you want to pass for parsing the lcls_elements.csv file.
Since generate.py had already parsed the csv file for other elements I figured we didn't need to duplicate the code here. Idk what your thoughts on this are.